plugins/ui-update/components/UIDetails.tsx

```diff
@@ -0,0 +1,217 @@
+import * as React from "react";
+import { componentFromStream, graphqlObservable } from "@dcos/data-service";
+import {
+  catchError,
+  delay,
+  filter,
+  map,
+  startWith,
+  switchMap,
+  take
+} from "rxjs/operators";
+import { BehaviorSubject, combineLatest, merge, Observable, of } from "rxjs";
+import gql from "graphql-tag";
+
+import ConfigurationMapSection from "#SRC/js/components/ConfigurationMapSection";
+import { schema as cosmosSchema } from "#SRC/js/data/cosmos";
+import { Package } from "#SRC/js/data/cosmos/Package";
+import { default as uiServiceSchema } from "#SRC/js/data/ui-update";
+import {
+  DEFAULT_UI_METADATA,
+  UIMetadata
+} from "#SRC/js/data/ui-update/UIMetadata";
+
+import AvailableUpdateRow from "#PLUGINS/ui-update/components/AvailableUpdateRow";
+import InstalledVersion from "#PLUGINS/ui-update/components/InstalledVersion";
+import InitialScreen from "#PLUGINS/ui-update/components/InitialScreen";
+import FallbackScreen from "#PLUGINS/ui-update/components/FallbackScreen";
+
+import {
+  EMPTY_ACTION,
+  UIAction,
+  UIActions,
+  UIActionType
+} from "#PLUGINS/ui-update/types/UIAction";
+import { versionUpdateAvailable } from "#PLUGINS/ui-update/utils";
+
+interface MutationError {
+  message: string;
+  name: string;
+}
+interface UIUpdateResult {
+  data: { updateDCOSUI: string };
+}
+interface UIRollbackResult {
+  data: { resetDCOSUI: string | null };
+}
+
+const uiServiceActions$ = new BehaviorSubject<UIAction>(EMPTY_ACTION);
```

Comment:
Or an `AsyncSubject` but that would also require some other changes to the `InstalledVersion` component since I'm relying on `EMPTY_ACTION` vs `null` `uiAction` prop for different rendering logic.

Reply:
ok! i'll mark that comment as resolved - just wanted to make sure that empty action serves a purpose, as i did not really understand it. :)